sympy/vector/coordsysrect.py

```diff
@@ -367,35 +335,33 @@ def _connect_to_standard_cartesian(cls, curv_coord_type, inverse=True):
             else:
                 inverse_transformation_equations = None
 
-        elif isinstance(curv_coord_type, collection.Callable):
-            pass
-            # TODO add code here
-            transformation_equations = Lambda((x1, x2, x3), curv_coord_type(x1, x2, x3))
-        
-        elif isinstance(curv_coord_type, (tuple, list, Tuple, collections.Callable)):
-            if variables is None:
-                variables = _default_variables('general')
-
-            if len(curv_coord_type) == 3:
-                transformation_equations = curv_coord_type
-                lame_coefficients = \
-                    cls._calculate_lame_coefficients(curv_coord_type)
+        else:
+            if isinstance(curv_coord_type, collections.Callable):
+                transformation_equations = Lambda((x1, x2, x3), curv_coord_type(x1, x2, x3))
+
+            elif isinstance(curv_coord_type, (tuple, list, Tuple)):
+                if len(curv_coord_type) == 3:
+                    transformation_equations = \
+                        Lambda((x1, x2, x3), tuple(i.subs(list(zip(variables, (x1, x2, x3))))
+                                                   for i in curv_coord_type))
+                else:
+                    raise ValueError("Wrong set of parameter.")
+
             else:
                 raise ValueError("Wrong set of parameter.")
 
+            if not cls._check_orthogonality(curv_coord_type, variables):
+                raise ValueError("The transformation equation does not "
+                                 "create orthogonal coordinate system")
+
+            lame_coefficients = cls._calculate_lame_coefficients(transformation_equations(*variables), variables)
```

Comment:
do we really need to pass `variables` here? I believe a single `lambda` is enough.